cmd/substation/vet.go

```diff
@@ -0,0 +1,194 @@
+	"regexp"
+
+	"github.com/brexhq/substation/v2"
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	rootCmd.AddCommand(vetCmd)
+	vetCmd.PersistentFlags().BoolP("recursive", "R", false, "recursively vet all files")
+	vetCmd.PersistentFlags().StringToString("ext-str", nil, "set external variables")
+}
+
+// vetTransformRe captures the transform ID from a Substation error message.
+// Example: `transform 324f1035-10a51b9a: object_target_key: missing required option` -> `324f1035-10a51b9a`
+var vetTransformRe = regexp.MustCompile(`transform ([a-f0-9-]+):`)
+
+var vetCmd = &cobra.Command{
+	Use:   "vet [path]",
+	Short: "report config errors",
+	Long: `'substation vet' reports errors in configuration files.
+
+The 'recursive' flag can be used to vet all files in a 
+directory, and the current directory is used if no arg is 
+provided.
+
+If an error is found, then the output always includes the
+file path and error message. If the location of the error
+is known, then the output also includes the line number
+where the error occurred.
+
+'vet' checks for two types of errors:
+  - Jsonnet syntax errors
+  - Substation configuration errors
+
+Jsonnet syntax errors look like this, and include the line
+number and column range where the error occurred:
+  vet.jsonnet:19:36-38 Unknown variable: st
+
+    sub.tf.obj.insert({obj: { trg: st.format('%s.-1', 'bar') }, value: 'baz'}),
+
+Substation config errors look like this, and include the
+line number where the error occurred in the 'transforms'
+array:
+  vet.jsonnet:3 transform 324f1035-10a51b9a: object_target_key: missing required option
+
+    {"type":"hash_sha256","settings":{"id":"324f1035-10a51b9a","object":{"source_key":"foo"}}}
+`,
+	// Examples:
+	//  substation vet [-R]
+	//  substation vet [-R] /path/to/configs
+	//  substation vet /path/to/config.json
+	//  substation vet /path/to/config.jsonnet
+	//  substation vet /path/to/my.libsonnet
+	Example: `  substation vet [-R]
+  substation vet [-R] /path/to/configs
+  substation vet /path/to/config.json
+  substation vet /path/to/config.jsonnet
+  substation vet /path/to/my.libsonnet
+`,
+	Args: cobra.MaximumNArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		// Default to current directory if no path is provided.
+		path := "."
+		if len(args) > 0 {
+			path = args[0]
+		}
+
+		// Catches an edge case where the user is looking for help.
+		if path == "help" {
+			fmt.Printf("warning: \"%s\" matched no files\n", path)
+			return nil
+		}
+
+		extStr, err := cmd.PersistentFlags().GetStringToString("ext-str")
+		if err != nil {
+			return err
+		}
+
+		recursive, err := cmd.Flags().GetBool("recursive")
+		if err != nil {
+			return err
+		}
+
+		return vetPath(path, extStr, recursive)
+	},
+}
+
+func vetFile(arg string, extVars map[string]string) error {
+	// This uses the custom config from the `test` command.
+	var cfg customConfig
+
+	switch filepath.Ext(arg) {
+	case ".jsonnet", ".libsonnet":
+		mem, err := buildFile(arg, extVars)
```

Comment:
Is this change meant to land after the open PR for `build`? I'd expect an error with the rename to `compileFile` since you're utilizing the return signature for that function with the config in memory vs the built `config.json` file.

Reply:
Yeah, this needs to be rebased.